### PR TITLE
Gpio/update

### PR DESCRIPTION
**Pull request by vireshk**
gpiod: Migrate to upstreamed version of libgpiod
    
    The libgpiod rust bindings got upstreamed recently. Migrate to the
    upstreamed version.

### CODE REVIEW

**Comment by vireshk** [comment]
@andreeaflorescu Some issues with builds here, timing out, any idea ?

**Comment by andreeaflorescu** [comment]
There seems to be some problem with the hosts on which we run the CIs and the first jobs that run will always timeout because they wait for the container to download. Even if the container is downloaded though the jobs take significantly longer to execute. @aghecenco did an initial investigation and it seems like all the threads are busy with epoll for some reason. We don't have a lot more details now, but we will prioritize this to find a fix. In the meantime, you can just retrigger the job and it should pass. This is not ideal, we'll keep you posted related to the fix.

**Comment by andreeaflorescu** [comment]
There is a legit failure from `cargo audit`. We should open an issue to track that.

**Comment by aghecenco** [comment]
> There is a legit failure from `cargo audit`. We should open an issue to track that.

See https://github.com/rust-vmm/rust-vmm-ci/issues/114 for the `audit` failure.

**Comment by vireshk** [comment]
> In the meantime, you can just retrigger the job and it should pass. This is not ideal, we'll keep you posted related to the fix.

Tried. Still not passing.

**Comment by andreeaflorescu** [comment]
We're tracking the timeout issue here: https://github.com/rust-vmm/community/issues/137

File: coverage_config_x86_64.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "coverage_score": 69.6,
+  "coverage_score": 68.9,
```
Comment (stsquad) [inline coverage_config_x86_64.json:2]: Is there any way to see which new code isn't covered by test cases?

**Review by stsquad — APPROVED** [review]

**Comment by vireshk** [comment]
@stefano-garzarella @mathieupoirier 

Can someone please review this ?

**Review by stefano-garzarella — APPROVED** [review]

**Pull request merged**